v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstovectorsearch/BigtableChangeStreamsToVectorSearch.java

```diff
@@ -0,0 +1,235 @@
+                            Utils.parseColumnMapping(options.getAllowRestrictsMappings()),
+                            Utils.parseColumnMapping(options.getDenyRestrictsMappings()),
+                            Utils.parseColumnMapping(options.getIntNumericRestrictsMappings()),
+                            Utils.parseColumnMapping(options.getFloatNumericRestrictsMappings()),
+                            Utils.parseColumnMapping(options.getDoubleNumericRestrictsMappings())))
+                    .withOutputTags(
+                        ChangeStreamMutationToDatapointOperationFn.UpsertDatapointTag,
+                        TupleTagList.of(
+                            ChangeStreamMutationToDatapointOperationFn.RemoveDatapointTag)));
+    results
+        .get(ChangeStreamMutationToDatapointOperationFn.UpsertDatapointTag)
+        .apply(WithKeys.of("foo")) // TOOD(meagar): Better name for place-holders?
+        .apply(
+            "Batch Contents",
+            GroupIntoBatches.<String, IndexDatapoint>ofSize(
+                    bufferSizeOption(options.getUpsertMaxBatchSize()))
+                .withMaxBufferingDuration(
+                    bufferDurationOption(options.getUpsertMaxBufferDuration()))
+            // .withShardedKey() // TODO(meagar): What does this do?
+            )
+        .apply("Values", Values.create())
+        .apply(
+            "Upsert Datapoints to VectorSearch",
+            ParDo.of(new UpsertDatapointsFn(vertexEndpoint, indexName)));
+
+    results
+        .get(ChangeStreamMutationToDatapointOperationFn.RemoveDatapointTag)
+        .apply(WithKeys.of("foo")) // TOOD(meagar): Better name for place-holders?
+        .apply(
+            "Batch Contents",
+            GroupIntoBatches.<String, String>ofSize(
+                    bufferSizeOption(options.getDeleteMaxBatchSize()))
+                .withMaxBufferingDuration(
+                    bufferDurationOption(options.getDeleteMaxBufferDuration()))
+            // .withByteSize() // TODO(meagar): We may also want to buffer by size
+            // .withShardedKey() // TODO(meagar): What does this do?
+            )
+        .apply("Values", Values.create())
+        .apply(
+            "Remove Datapoints From VectorSearch",
+            ParDo.of(new RemoveDatapointsFn(vertexEndpoint, indexName)));
+
+    return pipeline.run();
+  }
+
+  private static String getBigtableProjectId(BigtableChangeStreamsToVectorSearchOptions options) {
+    return StringUtils.isEmpty(options.getBigtableReadProjectId())
+        ? options.getProject()
+        : options.getBigtableReadProjectId();
+  }
+
+  private static Instant toInstant(Timestamp timestamp) {
+    if (timestamp == null) {
+      return null;
+    } else {
+      return Instant.ofEpochMilli(timestamp.getSeconds() * 1000 + timestamp.getNanos() / 1000000);
+    }
+  }
+
+  private static int bufferSizeOption(int size) {
+    if (size < 1) {
+      // TODO(meagar): Maybe throw instead?
```

Comment:
Not sure if it's idiomatically better to supply a correct value, or if I should throw something like an IllegalArgumentException